### DISCUSSION TITLE
how to write cypress e2e tests for supabase's magic link login authentication (react)?

### DISCUSSION THREAD

**Discussion by felibunnyy** [supabase/supabase#7839] [General]
Anyone knows how to write cypress e2e tests for retrieving the magic link after keying in the email field?

I cant seem to go beyond the login page since I cant automate the authentication, as this requires going to the user's keyed-in email to retrieve the unique magic link
<img width="513" alt="Screenshot 2022-07-21 at 11 28 22 PM" src="https://user-images.githubusercontent.com/105587981/180253130-c898dfa1-62da-4610-98d1-5ac453486bd8.png">
.

**Comment by EskelCz** [comment]
I'm new to Cypress but running into this exact issue. So far I've found this article, which seems like it might do try trick:
https://www.cypress.io/blog/2021/05/24/full-testing-of-html-emails-using-ethereal-accounts/
Reading the login email from supabase a logging in with it. The issue is, that login token should then ideally be preserved for all tests. I'm thinking about saving it into a variable and then saving it into localStorage before each test.

**Reply by EskelCz** [reply]
Update: Manually clicking the email is doable but really slow and least to weird bugs of cypress. So I went looking for some other approach and found this discussion: https://github.com/supabase/supabase/discussions/6177
It's not using the magic link but a service account from cypress node server. That works way better.

**Comment by mikeschneiderdotme** [comment]
Currently facing the same issue, I am running a local instance of supabase in Docker and magic links are sent to Inbucket email testing service that is running in parallel. I'm new to cypress and have not found a solution outside of getting the magic link url with the token and hardcoding it into the spec for the test to visit.

**Reply by galaxyblur** [reply]
Were you able to figure this out? I'm just getting start w/ Cypress integration, but I know there is an Inbucket JS client that easily talks to the Inbucket REST API and can fetch the contents of the email. (I got this working with Playwright.) Let me know where you ended up!
